compiler/dyno/lib/util/subprocess.cpp

```diff
@@ -28,61 +28,85 @@
 #include <sys/wait.h>
 #include <unistd.h>
 
-
 namespace chpl {
 
 
 // TODO: body of this function should call llvm::sys::ExecuteAndWait instead
 // see: https://llvm.org/doxygen/namespacellvm_1_1sys.html#a67688ad4697f1d516a7c71e41d78c5ba
-int executeAndWait(const std::vector<std::string> commandVec,
-             std::string description,
-             pid_t&      childPidOut,
-             bool        ignoreStatus,
-             bool        quiet,
-             bool        printSystemCommands) {
+int executeAndWait(const std::vector<std::string>& commandVec,
+                   const std::string& description,
+                   bool printSystemCommands) {
 
   // if an empty command passed, do nothing
-  if (commandVec.empty()) {
+  if (commandVec.empty() || commandVec[0].empty()) {
     return 0;
   }
 
-  // Join the elements of commandVec into a single space separated string
-  std::string commandStr = commandVec.front();
-  for (unsigned int i = 1; i < commandVec.size(); i++) {
-    commandStr += " " + commandVec.at(i);
+  // Treat a '#' at the start of a line as a comment
+  if (commandVec[0][0] == '#') {
```

Comment:
could this condition be moved into the conditional above since they all do the same thing and the short circuiting logic should protect reading an invalid index?